src/main/java/programminglife/controller/BookmarkController.java

```diff
@@ -0,0 +1,308 @@
+import java.util.List;
+
+/**
+ * Created by Martijn van Meerten on 15-5-2017.
+ * This class is a controller for loading storing and deleting bookmarks.
+ */
+public final class BookmarkController {
+    private static final String BOOKMARKPATH = "bookmarks.xml";
+
+    /**
+     * This class should not be instantiated.
+     * The functionality of this class does not rely on any particular instance.
+     * It relies solely on the contents of the bookmarks file.
+     */
+    private BookmarkController() { }
+
+    /**
+     * Default bookmark loading method.
+     * Uses the default bookmark path.
+     * @param graphName is the name of the graph int which this bookmark exists.
+     * @param bookMarkName is the name of the bookmark (needs to be unique for bookmarks in the same graph).
+     * @return the {@link Bookmark} you are trying to load.
+     */
+    public static Bookmark loadBookmark(String graphName, String bookMarkName) {
+        return loadBookmark(BOOKMARKPATH, graphName, bookMarkName);
+    }
+
+    /**
+     * Default all bookmark loading method.
+     * Uses the default bookmark path
+     * @param graphName The name of the graph from which to get the bookmarks.
+     * @return A List containing all bookmarks.
+     */
+    public static List<Bookmark> loadAllGraphBookmarks(String graphName) {
+        return loadAllGraphBookmarks(BOOKMARKPATH, graphName);
+    }
+
+    /**
+     * Default bookmark storing method.
+     * Uses the default bookmark path.
+     * @param graphName The name of the graph file.
+     * @param bookMarkName The name of the bookmark (must be unique for the graph).
+     * @param description The description of this bookmark.
+     * @param nodeID The ID of the node for searching.
+     * @param radius The radius of the bookmark for searching.
+     */
+    public static void storeBookmark(String graphName, String bookMarkName,
+                                     String description, int nodeID, int radius) {
+        storeBookmark(BOOKMARKPATH, graphName, bookMarkName, description, nodeID, radius);
+    }
+
+    /**
+     * Default bookmark storing method.
+     * Uses the default bookmark path.
+     * @param graphName The name of the graph file.
+     * @param bookMarkName The name of the bookmark (must be unique for the graph).
+     */
+    public static void deleteBookmark(String graphName, String bookMarkName) {
+        deleteBookmark(BOOKMARKPATH, graphName, bookMarkName);
+    }
+
+    /**
+     * Load a bookmark based on the graph's name and bookmark's name.
+     * @param fileName is the path of the bookmarks file location
+     * @param graphName is the name of the graph in which this bookmark exists.
+     * @param bookMarkName is the name of the bookmark (needs to be unique for bookmarks in the same graph).
+     * @return the {@link Bookmark} you are trying to load.
+     */
+    public static Bookmark loadBookmark(String fileName, String graphName, String bookMarkName) {
```

Comment:
This method is currently not used and will likely be removed, since `loadAllGraphBookmarks` is used to load the boakmarks